prov/shm/src/smr_ep.c

```diff
@@ -1354,6 +1355,11 @@ static int smr_ep_ctrl(struct fid *fid, int command, void *arg)
 		else
 			ep->smr_progress_ipc_list = smr_progress_ipc_list_noop;
 
+		if (smr_env.use_xpmem)
+			ep->region->xpmem_cap_self = SMR_VMA_CAP_ON;
+		else
+			ep->region->xpmem_cap_self = SMR_VMA_CAP_OFF;
+
```

Comment:
Can we move this setting into smr_create?
Also do we need to force disable xpmem if FI_HMEM was requested like for CMA? (see this right above in this file, line 1344)

Reply:
we do not need to disable xpmem in case of FI_HMEM. They can coexist. In fact we can memcpy between xpmem exported memory and device memory. I've tested this case.